l10n_th_account_tax_filing/models/account_tax_filing.py

```diff
@@ -0,0 +1,405 @@
+                )
+            )
+        if self.amount_to > 0:
+            invoice_lines.append(
+                (
+                    0,
+                    0,
+                    self.prepare_invoice_line(
+                        self.account_to_id, self.amount_to, sign * -1
+                    ),
+                )
+            )
+        return invoice_lines
+
+    def create_invoice(self, move_type):
+        # Create invoice
+        invoice = {
+            "move_type": move_type,
+            "partner_id": self.partner_id.id,
+            "invoice_date": fields.Date.today(),
+            "tax_filing_id": self.id,
+            "invoice_line_ids": self.create_invoice_line(move_type),
+        }
+        return invoice
+
+    def prepare_account_move_line(self, account, debit, credit):
+        return {
+            "name": account.name,
+            "account_id": account.id,
+            "partner_id": self.partner_id.id,
+            "debit": debit,
+            "credit": credit,
+            "is_tax_filing": True,
+        }
+
+    def create_account_move_line(self):
+        amount = self.amount_from - self.amount_to
+        lines = []
+        if self.amount_from > 0:
+            lines.append(
+                (
+                    0,
+                    0,
+                    self.prepare_account_move_line(
+                        self.account_from_id, self.amount_from, 0
+                    ),
+                )
+            )
+        if self.amount_to > 0:
+            lines.append(
+                (
+                    0,
+                    0,
+                    self.prepare_account_move_line(
+                        self.account_to_id, 0, self.amount_to
+                    ),
+                )
+            )
+        if self.has_adjustment and amount != 0:
+            lines.append(
+                (
+                    0,
+                    0,
+                    self.prepare_account_move_line(
+                        self.account_adjust_id,
+                        0 if amount > 0 else abs(amount),
+                        abs(amount) if amount > 0 else 0,
+                    ),
+                )
+            )
+        return lines
+
+    def create_journal_entry(self):
+        journal_entry = {
+            "move_type": "entry",
+            "ref": self.name,
+            "invoice_date": fields.Date.today(),
+            "tax_filing_id": self.id,
+            "line_ids": self.create_account_move_line(),
+        }
+        return journal_entry
+
+    def action_view_entries(self):
+        return {
+            "name": _("Journal Entries"),
+            "view_mode": "form",
+            "res_model": "account.move",
+            "res_id": self.move_id.id,
+            "type": "ir.actions.act_window",
+        }
+
+    def action_confirm(self):
+        for record in self:
+            if not record.line_ids:
+                raise UserError(_("Cannot confirm tax filing without journal items"))
```

Comment:
```suggestion
                raise UserError(_("You need to add a line before confirm."))
```

Error should read easy understand, why error and how we can solved it.